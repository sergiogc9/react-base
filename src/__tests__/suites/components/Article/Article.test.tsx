import { shallow } from "enzyme";
import React from "react";
import { RouterState } from 'react-router-redux';
import queryString from 'query-string';

import Article from "@src/components/Article/Article";

const location: RouterState["location"] = {
	pathname: "/article",
	search: "?period=last_week",
	state: {},
	hash: ""
};

let onSetFormAndFilters: any;

describe("<Article />", () => {
	let wrapper: any;

	beforeEach(() => {

		onSetFormAndFilters = jest.fn();

		wrapper = shallow(<Article
			location={location}
			onSetFormAndFilters={onSetFormAndFilters}
		/>);
	});

	it("Test component snapshot", () => {
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Test on set form and filters execution", () => {
		expect(onSetFormAndFilters).toHaveBeenCalledTimes(1);
		expect(onSetFormAndFilters).toHaveBeenCalledWith(queryString.parse(location.search));
	});

});
