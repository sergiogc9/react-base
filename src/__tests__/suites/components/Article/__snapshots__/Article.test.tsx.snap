// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> Test component snapshot 1`] = `
"<div id=\\"article\\">
  <Connect(ArticleFacets) />
  <div className=\\"main-drawer\\">
    <Connect(ArticleForm) />
    <Connect(ArticleActions) />
    <Connect(SearchResults) />
    <Connect(SearchPaginator) />
  </div>
</div>"
`;
